### PR TITLE
Analytics

### LINKED ISSUE #115: Add analytics
It may be too early to think much about this, but something that will be really helpful for improving how people use the app is installing some analytics to see what pages and routes users are using most, and gain UX insights from. The Atom team uses Google Analytics and a [custom package](https://github.com/atom/metrics) for this, which also has the benefit of providing aggregate stats about how many total users there are for the app, and what their activity on a weekly/monthly basis is like.

### PR DESCRIPTION
**Pull request by stukalin**
Fixes #115 .

**Changes proposed in this pull request:**
- Google Analytics via `ember-metrics` library
- offline mode support with a lib from Google

cc @HospitalRun/core-maintainers

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -7,6 +7,23 @@ module.exports = function(environment) {
     environment: environment,
     rootURL: process.env.EMBER_CLI_ELECTRON ? null : '/',
     locationType: 'hash', // Using hash location type because it is more friendly for offline.
+    metricsAdapters: [
+      {
+        name: 'GoogleAnalytics',
+        environments: ['development', 'production'],
+        config: {
+          id: 'UA-TODO-HOSPITAL-RUN',
```
Comment (stukalin) [inline config/environment.js]: @tangollama we'd need the ID here

File: package.json [hunk 2]
```diff
@@ -114,8 +115,8 @@
   "dependencies": {
     "electron-compile": "^6.4.2",
     "electron-localshortcut": "^3.0.4",
-    "electron-protocol-serve": "1.3.0",
```
Comment (stukalin) [inline package.json]: that was duplicated somehow

File: app/router.js [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
+    this._super(...arguments);
+    this._trackPage();
+  },
+
+  _trackPage() {
+    scheduleOnce('afterRender', this, () => {
+      let page = this.get('url');
+      let title = this.getWithDefault('currentRouteName', 'unknown');
+
+      get(this, 'metrics').trackPage({ page, title });
```
Comment (stukalin) [inline app/router.js]: Well, this piece of code was copy-pasted from the `ember-metrics` sample. Apparently we'd need something more sophisticated

**Comment by stukalin** [comment]
@jglovier it's surely the former and yes, you're right, it's confusing now which GA account should be used. It also allows everyone to change the config easily, so maybe some encapsulation of GA-ID would be required there.

**Comment by stukalin** [comment]
So, I see 3 solutions there:
1. If we really say "Oh, no one really edits `config/environment` on their own!", let's leave it as it is.
2. We would be able to use [Lazy initialization](https://github.com/poteto/ember-metrics#lazy-initialization) in order to encapsulate _a bit_ the ID and put it in our `route.js`.
3. And if we really care that our ID is... well... call it safe or maybe not-so-confusing, then the possible solution could be creating a custom `hr-google-analytics` metrics adapter and encapsulate the ID there.

@jglovier @tangollama @jkleinsc opinions?

**Comment by jglovier** [comment]
@stukalin thanks for the response. @jkleinsc @tangollama I'd be interested to hear your thoughts on ☝️

**Pull request merged**